students/zach_meves/lesson03/basic_operations.py

```diff
@@ -0,0 +1,96 @@
+"""
+Basic database operations.
+"""
+
+from customer_model import Customer, DB
+
+
+def add_customer(customer_id: int, name: str, lastname: str, home_address: str,
+                 phone_number: str, email_address: str, status: bool, credit_limit: float):
+    """
+    Add a new customer to the database.
+
+    :param customer_id: int, Customer ID
+    :param name: str, First name
+    :param lastname: str, Last name
+    :param home_address: str, Home address
+    :param phone_number: str, Phone number
+    :param email_address: str, Email address
+    :param status: bool, True for "active", False for "inactive"
+    :param credit_limit: float, credit limit
+    """
+
+    # Create a new customer object
+    with DB.transaction():
+        try:
+            cust = Customer.create(id=customer_id, name=name, last_name=lastname,
+                                   address=home_address, phone=phone_number, email=email_address,
+                                   status=status, credit_limit=credit_limit)
+        except Exception as err:
+            raise ValueError(err)
+        else:
+            cust.save()
+
+
+def search_customer(customer_id: int) -> dict:
+    """
+    Return a dictionary representing the customer with the given ID.
+    :param customer_id: int, customer ID
+    :return: dict
+    """
+
+    try:
+        res = _get_by_id(customer_id)
+    except Exception:
```

Comment:
Avoid using general exceptions; the risk being that new issues not seen in the development process might be caught by _except Exception_, masking and improperly handling new issues. Instead, replace this block with specific exception types.